vector/v.db.select/main.c

```diff
@@ -305,44 +378,60 @@ int main(int argc, char **argv)
 
             db_convert_column_value_to_string(column, &value_string);
 
-            if (!flags.colnames->answer && flags.vertical->answer)
+            if (!flags.colnames->answer && vertical)
                 fprintf(stdout, "%s%s", db_get_column_name(column), fsep);
 
-            if (col && !flags.vertical->answer && !flags.json->answer)
+            if (col && !vertical && !json)
                 fprintf(stdout, "%s", fsep);
 
-            if (flags.json->answer) {
+            if (json) {
                 if (!col)
                     fprintf(stdout, "{");
                 fprintf(stdout, "\"%s\":", db_get_column_name(column));
             }
 
-            if (options.nullval->answer && db_test_value_isnull(value)) {
-                if (flags.json->answer)
-                    fprintf(stdout, "\"%s\"", db_get_column_name(column));
-                else
+            if (db_test_value_isnull(value)) {
+                if (json)
```

Comment:
In this case, we shouldn't allow `null_value=` with `format=json` or we need to output `"\"%s\"", options.nullval->answer` for JSON.

Reply:
Fixed in 334dacda2. I think not allowing is the correct choice here assuming you can do whatever you want once your read the JSON.